### PR TITLE
Mc reviews

### PR DESCRIPTION
**Pull request by matt-collins087**
Added some sorting functionality in reviewsList
Added functionality to dynamically render reviews on post
added css styling for components

### CODE REVIEW

File: client/src/components/Overview/AddToCart 2.jsx [hunk 1]
```diff
@@ -0,0 +1,118 @@
+import React, { useState, useEffect} from 'react';
```
Comment (stevehackreactor) [inline client/src/components/Overview/AddToCart 2.jsx]: This file is called AddToCart 2. It looks like it is my AddToCart component so I'm wondering why it is called AddToCart 2 and if it is a duplicate file.

File: client/src/components/Overview/ProductDetails 2.jsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import React, {useState, useEffect} from 'react';
```
Comment (stevehackreactor) [inline client/src/components/Overview/ProductDetails 2.jsx]: It looks like the same situation with Product Details and Style Selector potentially being duplicates.

File: client/src/components/Overview/StyleSelector 2.jsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
+import React, {useState, useEffect} from 'react';
+
```
Comment (stevehackreactor) [inline client/src/components/Overview/StyleSelector 2.jsx]: Potentially a duplicate

File: client/src/components/ReviewsRatings/NewReview.jsx [hunk 4]
```diff
@@ -59,11 +58,10 @@ const NewReview = ({ reviews, productId, reviewChar }) => {
...
   };
 
+
+
   return (
     <div>
-      <Button variant="outline-dark">More Reviews</Button>
-      <Button variant="outline-dark" onClick={handleShow}>Add a Review +</Button>
-
       <Modal show={show} onHide={handleClose} centered>
```
Comment (stevehackreactor) [inline client/src/components/ReviewsRatings/NewReview.jsx]: Add some white space on either side of the params to show/onHide for visual consistency.

File: client/src/components/Overview/AddToCart 2.jsx (hunk 1 above)
Comment (matt-collins087) [reply client/src/components/Overview/AddToCart 2.jsx]: Dont know how that happened. Just deleted it.

File: client/src/components/Overview/ProductDetails 2.jsx (hunk 2 above)
Comment (matt-collins087) [reply client/src/components/Overview/ProductDetails 2.jsx]: Dont know how that happened. Just deleted it.

File: client/src/components/Overview/StyleSelector 2.jsx (hunk 3 above)
Comment (matt-collins087) [reply client/src/components/Overview/StyleSelector 2.jsx]: Dont know how that happened. Just deleted it.

File: client/src/components/ReviewsRatings/RatingsReviews.jsx [hunk 5]
```diff
@@ -24,21 +26,30 @@ const RatingsReviews = ({ productId, changeId, reviews, changeReviews }) => {
...
+    setCount(2);
+  }, [productId, newReview]);
 
   return (
     <div className="ratings-reviews-container">
 
       {(reviews.length > 0 && reviewMetaData) ?
-        <div className="ratings-container">
+        <div>
           <RatingsList reviews={reviews} reviewMetaData={reviewMetaData}/>
```
Comment (stevehackreactor) [inline client/src/components/ReviewsRatings/RatingsReviews.jsx]: Add spacing around params reviews and reviewMetaData

File: client/src/components/ReviewsRatings/RatingsReviews.jsx [hunk 6]
```diff
@@ -24,21 +26,30 @@ const RatingsReviews = ({ productId, changeId, reviews, changeReviews }) => {
...
         </div>
-        : null
+        : <div></div>
       }
-      {reviewMetaData ?
-        <div className="reviews-container">
-          <ReviewList reviews={reviews} productId={productId} reviewChar={reviewMetaData.characteristics} />
```
Comment (stevehackreactor) [inline client/src/components/ReviewsRatings/RatingsReviews.jsx:41]: Add spacing again.

File: client/src/components/ReviewsRatings/Review.jsx [hunk 7]
```diff
@@ -16,24 +16,33 @@ const Review = ({ review }) => {
...
       </div>
       <h3>{review.summary}</h3>
       <p>{review.body}</p>
+      {(review.response !== null && review.response.length > 0) ?
+        <div className="review-response">
+          <h5>Response:</h5>
+          <p>{review.response}</p>
+        </div>
+        : null}
       {review.photos.map((photo, index) => {
```
Comment (stevehackreactor) [inline client/src/components/ReviewsRatings/Review.jsx:38]: Add spacing around params

File: client/src/components/ReviewsRatings/ReviewList.jsx [hunk 8]
```diff
@@ -1,16 +1,114 @@
...
+          productId={productId}
+          reviewChar={reviewChar}
+          reviews={reviews}
+          newReview={newReview}
+          setNewReview={setNewReview}
+          show={show}
+          setShow={setShow}
+          loadReviews={loadReviews}/>
+      </div>
+      <Button id="more-reviews-button" className="review-button" variant="info" onClick={updateCount}>More Reviews</Button>
```
Comment (stevehackreactor) [inline client/src/components/ReviewsRatings/ReviewList.jsx]: spacing on params

**Review by stevehackreactor — CHANGES_REQUESTED** [review]
@matt-collins087 I have made comments mainly on styling on readability. Other than that looks great!

**Pull request merged**